src/sentry/models/user.py

```diff
@@ -356,27 +356,38 @@ def merge_to(from_user, to_user):
                 except IntegrityError:
                     pass
 
-        model_list = (
+        fk_model_list = (
             Authenticator,
+            Identity,
+            UserAvatar,
+            UserEmail,
+            UserOption,
+        )
+
+        non_fk_model_list = (
             GroupAssignee,
             GroupBookmark,
             GroupSeen,
             GroupShare,
             GroupSubscription,
-            Identity,
-            UserAvatar,
-            UserEmail,
-            UserOption,
         )
 
-        for model in model_list:
+        for model in fk_model_list:
             for obj in model.objects.filter(user=from_user):
                 try:
                     with transaction.atomic():
                         obj.update(user=to_user)
                 except IntegrityError:
                     pass
 
+        for model in non_fk_model_list:
+            for obj in model.objects.filter(user_id=from_user.id):
+                try:
+                    with transaction.atomic():
+                        obj.update(user_id=to_user.id)
+                except IntegrityError:
+                    pass
+
```

Comment:
I wonder if we're better off just keeping these in the same list, and always just querying by and saving to `user_id`. Saves you needing to update this in later prs.

Reply:
Yup, I think originally there was a reason to split the two but in the end there isn't.